share/wake/lib/core/examples.wake

```diff
@@ -0,0 +1,134 @@
+# Copyright 2021 SiFive, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You should have received a copy of LICENSE.Apache2 along with
+# this software. If not, you may obtain a copy at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+package wake
+
+# Ported from the jq tutorial
+export def example1 (j: JValue): Result JValue Error = Pass j
+export def example2 (j: JValue): Result JValue Error =
+    j | jAt 0 | expectOne "j was empty"
+
+# Simplify github json first commit
+export def example3 (log: JValue): Result JValue Error =
+    require Pass commit =
+        log
+        | jAt 0
+        | expectOne "expected at least one commit"
+
+    require Pass message =
+        commit
+        | jField "commit" |> jField "message"
+        | expectOne ".commit.message missing"
+
+    require Pass name =
+        commit
+        | jField "commit" |> jField "name"
+        | expectOne ".commit.name missing"
+
+    JObject ("message" :-> message, "name" :-> name, Nil) | Pass
```

Comment:
Please format as:
```
def output = JObect (
  "message" ...,
  "name" ...,
  Nil
)

Pass output
```

... we want the examples to show the expected structure used in larger situations.

Reply:
I was going to delete these examples at the end of review, do you think we should put reviewed versions of them up somewhere?

Reply:
I think we should have "how to JSON" documentation. These examples (or something similar) belong in the document.